### PR TITLE
INTYGFV-15998: Include only relations to signed certificates.

### PR DESCRIPTION
**Pull request by mikeyOne**

### CODE REVIEW

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
Nice! A few minor refactoring comments to make the code (and test code) easier to maintain and read.

I think also we should add tests to verify that  LOCKED and  REVOKED,  LOCKED_REVOKED aren't included. I know the code caters for that right now, but there are no tests that verifies it. If someone changes the logic to !UNSIGNED all the tests will still pass.

File: integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/metadata/StatusConverter.java [hunk 1]
```diff
@@ -12,14 +13,15 @@
...
+import se.inera.intyg.minaintyg.integration.webcert.client.dto.metadata.CertificateStatus;
 
 @Component
 public class StatusConverter {
 
   public List<CertificateStatusType> convert(CertificateMetadataDTO metadataDTO) {
     final var events = new ArrayList<CertificateStatusType>();
 
-    if (isReplaced(metadataDTO)) {
+    if (isReplaced(metadataDTO) && replacingCertificateIsSigned(metadataDTO)) {
```
[the remark this answers is not in the document]
Comment (mikeyOne) [reply integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/metadata/StatusConverter.java:24]: done

File: integration-webcert/src/test/java/se/inera/intyg/minaintyg/integration/webcert/converter/metadata/StatusConverterTest.java [hunk 2]
```diff
@@ -133,6 +134,22 @@ void shallConvertToReplacedIfCertificateIsReplaced() {
...
+    @Test
+    void shallNotConvertToReplacedIfCertificateIsReplacedAndReplacingCertificateIsNotSigned() {
+      final var metadataDTO = CertificateMetadataDTO.builder()
+          .created(OLD_DATE)
+          .relations(
+              createChild(CertificateRelationType.REPLACED)
+          )
+          .build();
+
+      Arrays.stream(metadataDTO.getRelations().getChildren())
```
[the remark this answers is not in the document]
Comment (mikeyOne) [reply integration-webcert/src/test/java/se/inera/intyg/minaintyg/integration/webcert/converter/metadata/StatusConverterTest.java:146]: done

**Review by mww-sopra-steria — APPROVED** [review]
Looks great! Approved!

**Pull request merged**